src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlMetadataFactory.cs

```diff
@@ -51,11 +51,13 @@ private void addUDTsToDataTypesTable(DataTable dataTypesTable, SqlConnection con
 
             SqlCommand engineEditionCommand = connection.CreateCommand();
             engineEditionCommand.CommandText = "SELECT SERVERPROPERTY('EngineEdition');";
-            var engineEdition = (int)engineEditionCommand.ExecuteScalar()!;
+            var engineEdition = (int)engineEditionCommand.ExecuteScalar();
 
-            if (engineEdition == 9)
+            if (engineEdition is 6 or 9 or 11)
```

Comment:
Nit: This could be extracted into an array or list called `AssemblyPropertyUnsupportedEngines` for better readability.

Reply:
@saurabh500 PR updated with your suggestion